torch/onnx/_type_utils.py

```diff
@@ -0,0 +1,245 @@
+"""Utilities for converting and operating on ONNX, JIT and torch types."""
+from __future__ import annotations
+
+import enum
+from typing import Dict, Optional, Union
+
+from typing_extensions import Literal
+
+import torch
+from torch._C import _onnx as _C_onnx
+
+ScalarName = Literal[
+    "Byte",
+    "Char",
+    "Double",
+    "Float",
+    "Half",
+    "Int",
+    "Long",
+    "Short",
+    "Bool",
+    "ComplexHalf",
+    "ComplexFloat",
+    "ComplexDouble",
+    "QInt8",
+    "QUInt8",
+    "QInt32",
+    "BFloat16",
+    "Undefined",
+]
+
+TorchName = Literal[
+    "bool",
+    "uint8_t",
+    "int8_t",
+    "double",
+    "float",
+    "half",
+    "int",
+    "int64_t",
+    "int16_t",
+    "complex32",
+    "complex64",
+    "complex128",
+    "qint8",
+    "quint8",
+    "qint32",
+    "bfloat16",
+]
+
+
+class ScalarType(enum.IntEnum):
+    """Scalar types defined in torch."""
+
+    # Order defined in https://github.com/pytorch/pytorch/blob/344defc9733a45fee8d0c4d3f5530f631e823196/c10/core/ScalarType.h
```

Comment:
Is there a way to expose/reuse original api, rather than redefining?

Reply:
That's a good question. I haven't found a clear path. Do you have any suggestions?

Reply:
Mark as resolved: out of scope from our discussion